### PR TITLE
adding vue

### PR DESCRIPTION
**Pull request by mirzaakhena**
i'am adding vue js

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,15 +9,17 @@
   },
   "author": "equanox",
   "scripts": {
-    "install": "(cd app && npm install) && (cd ui/js && npm install) && (cd ui/react && npm install) && (cd ui/typescript && npm install)",
+    "install": "(cd app && npm install) && (cd ui/js && npm install) && (cd ui/react && npm install) && (cd ui/typescript && npm install) && (cd ui/vue && npm install)",
     "postinstall": "go get -d ./...",
     "clean": "(rm -r app/node_modules) && (rm -r ui/js/node_modules) && (rm -r ui/react/node_modules) && (rm -r ui/typescript/node_modules)",
```
Comment (Equanox) [inline package.json]: Cleanup ui/vue/node_modules

**Comment by Equanox** [comment]
Thanks for your PR. Happy to have vue.js on bord.
Can you update the main README.md accordingly, to give everybody a quickstart to vuejs?

**Comment by mirzaakhena** [comment]
I simplify the .gitignore 
remove app/assets/js/index.js since this file can be generated by npm run build-xxx
and i do some fixing in vue webpack.config

File: .gitignore [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
-vendor/
+node_modules/
+package-lock.json
+build/
+
+.vscode/
+
+*.d.ts
+
+app/assets/js/*
```
Comment (Equanox) [inline .gitignore]: vendor/ 

It's created when installing go dependencies using https://github.com/golang/dep

**Comment by Equanox** [comment]
I appreciate the removal of  *app/assets/js/index.js* but we need to make sure  *app/assets/js* exists or is created on the first run of `npm run build-xxx` otherwise we get a error. Either create a dummy placeholder in *app/assets/js* or create the folder in `npm run build-xxx` cmds.

File: .gitignore (hunk 2 above)
Comment (mirzaakhena) [reply .gitignore]: Sure. I will add vendor/ too

**Comment by mirzaakhena** [comment]
Ah yes.. i didnt realize removing the app/assets/js folder. 
And honestly i just knew that git does not track the folder, it track files! :))
I learn the trick to keep the folder but not the files [here](https://stackoverflow.com/questions/5091017/how-to-track-directories-but-not-their-files-with-git)

**Comment by Equanox** [comment]
Thanks, didn't knew this trick either. Good Job.

**Pull request merged**